web/src/server/api/routers/comments.ts

```diff
@@ -288,4 +261,104 @@ export const commentsRouter = createTRPCRouter({
+        sessionId: z.string(),
+        queryClickhouse: z.boolean().default(false),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      try {
+        return await measureAndReturnApi({
+          input,
+          operation: "comments.getTraceCommentCountsBySessionId",
+          user: ctx.session.user ?? undefined,
+          pgExecution: async () => {
+            const session = await ctx.prisma.traceSession.findFirst({
+              where: {
+                id: input.sessionId,
+                projectId: input.projectId,
+              },
+              include: {
+                traces: {
+                  select: {
+                    id: true,
+                  },
+                },
+              },
+            });
+            if (!session) {
+              throw new TRPCError({
+                code: "NOT_FOUND",
+                message: "Session not found in project",
+              });
+            }
+
+            const allCommentCounts = await ctx.prisma.$queryRaw<
+              Array<{ objectId: string; count: bigint }>
+            >`
+            SELECT object_id as "objectId", COUNT(*) as count
+            FROM comments
+            WHERE project_id = ${input.projectId}
+            AND object_type = 'TRACE'
+            GROUP BY object_id
+          `;
+
+            const traceIds = new Set(
+              session.traces.map((t: { id: string }) => t.id),
+            );
+            return new Map(
+              allCommentCounts
+                .filter((c) => traceIds.has(c.objectId))
+                .map(({ objectId, count }) => [objectId, Number(count)]),
+            );
+          },
+          clickhouseExecution: async () => {
+            const postgresSession = await ctx.prisma.traceSession.findFirst({
+              where: {
+                id: input.sessionId,
+                projectId: input.projectId,
+              },
+            });
+
+            if (!postgresSession) {
+              throw new TRPCError({
+                code: "NOT_FOUND",
+                message: "Session not found in project",
+              });
+            }
+
+            const clickhouseTraces = await getTracesIdentifierForSession(
+              input.projectId,
+              input.sessionId,
+            );
+
+            const allCommentCounts = await ctx.prisma.$queryRaw<
+              Array<{ objectId: string; count: bigint }>
+            >`
+              SELECT object_id as "objectId", COUNT(*) as count
+              FROM comments
+              WHERE project_id = ${input.projectId}
+              AND object_type = 'TRACE'
+              GROUP BY object_id
+            `;
+
+            const traceIds = new Set(clickhouseTraces.map((t) => t.id));
+            return new Map(
+              allCommentCounts
+                .filter((c) => traceIds.has(c.objectId))
+                .map(({ objectId, count }) => [objectId, Number(count)]),
+            );
+          },
+        });
```

Comment:
If one of our clickhouse experts could quickly review this please, that would be very helpful. Just want to make sure my implementation is correct for clickhouse. No need to check postgres, will likely never be used. 

Context: comment table is really small (don't want to share public stats here, will dm if needed)

Reply:
Quick q: why did you think that the comment table size matters? We look up CH only by session_id, correct? I dont want to miss something here.

Reply:
Was just an info re the allCommentCounts query, nothing clickhouse specific -- you did not miss anything :)